### DISCUSSION TITLE
Does kubeletstats receiver require K8S DNS server to resolve K8S cluster's node names?

### DISCUSSION THREAD

**Discussion by sshiba** [open-telemetry/opentelemetry-python-contrib#2465] [General]
I am experiencing an error with `kubeletstats` receiver where it is unable to resolve K8S cluster's node name. See error message below:

`2024-04-24T14:04:28.918Z	error	scraperhelper/scrapercontroller.go:197	Error scraping metrics	{"kind": "receiver", "name": "kubeletstats", "data_type": "metrics", "error": "Get \"https://<node-name>:10250/stats/summary\": dial tcp: lookup <node-name> on 10.10.0.10:53: no such host", "scraper": "kubeletstats"}`

Is there an option for the `kubeletstats` receiver to bypass the DNS server?

**Comment by pmcollins** [comment]
Hi @sshiba -- I think you'll want to move this to opentelemetry-collector-contrib. Cheers.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/README.md
